contracts/contracts/buyback/Buyback.sol

```diff
@@ -84,18 +87,35 @@ contract Buyback is Governable {
     }
 
     /**
-     * @dev Execute a swap of OGN for OUSD via Uniswap or Uniswap compatible
+     * @dev Execute a swap of OGV for OUSD via Uniswap or Uniswap compatible
      * protocol (e.g. Sushiswap)
      **/
     function swap() external onlyVault nonReentrant {
-        uint256 sourceAmount = ousd.balanceOf(address(this));
-        if (sourceAmount < 1000 * 1e18) return;
-        if (uniswapAddr == address(0)) return;
-        // 97% should be the limits of our oracle errors.
-        // If this swap sometimes skips when it should succeed, that’s okay,
-        // the amounts will get get sold the next time this runs,
-        // when presumably the oracles are more accurate.
-        uint256 minExpected = expectedOgnPerOUSD(sourceAmount).mul(97).div(100);
+        // Disabled for now, will be manually swapped by
+        // `strategistAddr` using `swapNow()` method
+
+        // TODO: should there be a isDisabled() function that
+        // the vault could use? Will need an contract upgrade
+        // if we make that change to the Vault
+        return;
+    }
+
+    /**
+     * @dev Execute a swap of OGV for OUSD via Uniswap or Uniswap compatible
+     * protocol (e.g. Sushiswap)
+     * @param ousdAmount OUSD to sell
+     * @param minExpected mininum amount of OGV to receive
+     **/
+    function swapNow(uint256 ousdAmount, uint256 minExpected)
+        external
+        onlyGovernorOrStrategist
+        nonReentrant
+    {
+        require(uniswapAddr != address(0), "Exchange address not set");
+        require(minExpected > 0, "Invalid minExpected value");
+
+        uint256 ousdBalance = ousd.balanceOf(address(this));
+        require(ousdBalance >= ousdAmount, "Insufficient OUSD balance");
 
         UniswapV3Router.ExactInputParams memory params = UniswapV3Router
             .ExactInputParams({
```

Comment:
We can make some changes to the path here and save money:

- The usdt/weth9 path can now use the 500 fee since there is lots of liquidity in that pool now, which saves us a lot on fees. 
- Origin removed the OGV/OGN pool and moved liquidity to an ETH/OGV pool. We'll want to use that pool for swapping. This also means we can remove tracking the OGN address from this contract.

Reply:
Have change it, thanks for the tip :)